### DISCUSSION TITLE
Recommended way to install biome in a monorepo

### DISCUSSION THREAD

**Discussion by salieflewis** [biomejs/biome#280] [Q&A] [answered]
Hi, I realize this may be a bit of a long discussion, but hopefully this conversation provides some answers for those who come after and are looking to do the same thing. If one were looking to install biome in a monorepo, would the recommendation be installing it as a dev dependency in the root of your project, and having a single `biome.json` that various `apps` and `packages` are configured from. Maybe the assumption I'm making here is that each app and package here is written in a language that biome supports and in fact wants linting or formatting applied to it.

Second question. What is the significance of `"files"` in your configuration file. I thought by ignoring certain files, this would opt them out from both linting and formatting, but that doesn't seem to be the case. Its still necessary to go in and configure those files to be ignored by both the linter and the formatter?

If you had different biome config hopes for various apps and packages, would you recommend creating a separate `biome.json` file inside those respective directories and then extending the child config to the parent? Is this possible? i.e. `"extends": ["../../biome.json"]`. Lastly, instead of ignore, is there a way to say, "only look at this folder" in the context of another directory?

Okay, realize this is super long, but think there are some valid questions here. Appreciate anyone willing to lend a hand.

**Comment by ematipico** [comment]
> What is the significance of "files" in your configuration file. I thought by ignoring certain files, this would opt them out from both linting and formatting, but that doesn't seem to be the case. Its still necessary to go in and configure those files to be ignored by both the linter and the formatter?

`files.ignore` should ignore all the files that match the patterns inside the array. If that's not the case, it's probably a bug. You can see how we do it inside [our repository](https://github.com/biomejs/biome/blob/276ffe3486c3ca318efcfaff068d3c86165ccf5c/biome.json#L8-L10)

> f you had different biome config hopes for various apps and packages, would you recommend creating a separate biome.json file inside those respective directories and then extending the child config to the parent? Is this possible? i.e. "extends": ["../../biome.json"]

Yes, that the suggested way. Consider that Biome also goes upwards the file system for a configuration file, if one isn't found in the working directory.

> Lastly, instead of ignore, is there a way to say, "only look at this folder" in the context of another directory?

Not yet, but some people asked to implement a `include` option, that works the opposite of `ignore`. I think it's worth having it.

**Reply by ematipico** [reply]
FYI, I am writing a small guide for these cases

**Reply by salieflewis** [reply]
Appreciate you! Happy to check it out when its ready :)

**Reply by ematipico** [reply, marked as answer]
https://biomejs.dev/guides/big-projects/
